lib/blemulator/simulation/simulated_characteristic.dart

```diff
@@ -36,18 +36,21 @@ class SimulatedCharacteristic {
 
   Future<void> write(Uint8List value) async {
     this._value = value;
-    if (_streamController?.hasListener == true)
-      _streamController.sink.add(value);
+    if (streamController?.hasListener == true)
+      streamController.sink.add(value);
   }
 
   Stream<Uint8List> monitor() {
-    if (_streamController == null) {
-      _streamController = StreamController(onCancel: () {
-        _streamController.close();
-        _streamController = null;
+    if (!isNotifiable) {
+      streamController.sink.addError("Characterisitc is not notifiable");
```

Comment:
NPE

Reply:
Also: this error will be consumed before the subscription takes place

Reply:
Also: if you cannot monitor this characteristic, then you should either fail in SimulationManager beforehand or return a stream with only an error.